macOS/DuckDuckGo/Preferences/Model/PreferencesSidebarModel.swift

```diff
@@ -266,12 +267,21 @@ final class PreferencesSidebarModel: ObservableObject {
 
         if subscriptionManager.isUserAuthenticated {
             // Calculate current user entitlements
-            var currentUserEntitlements: [Entitlement.ProductName] = []
-            let entitlements: [Entitlement.ProductName] = [.networkProtection, .dataBrokerProtection, .identityTheftRestoration, .identityTheftRestorationGlobal]
+            var currentUserEntitlements: [SubscriptionEntitlement] = []
+            let entitlements: [SubscriptionEntitlement] = [.networkProtection, .dataBrokerProtection, .identityTheftRestoration, .identityTheftRestorationGlobal]
 
-            for entitlement in entitlements {
-                if let hasEntitlement = try? await subscriptionManager.isEnabled(feature: entitlement), hasEntitlement {
+            if let subscriptionManagerV2 = subscriptionManager as? SubscriptionManagerV2,
```

Comment:
Updated check for entitlements by getting tokenContainer and checking them directly

Reply:
In this way, you are completely bypassing the subscription entitlements, making it useless to download and store them. They are there, especially for the UI, do you think they are redundant? 
How is this going to work when/if the subscription entitlements change? (new country, change in features etc?)

Reply:
They are fetched in macOS/DuckDuckGo/Preferences/Model/PreferencesSidebarModel.swift:L265.
I need these two sets (features provided by subscription and user's entitlements) separately as they are used to calculate different things.

Reply:
Got it, LGTM